app/api_v1/builds.py

```diff
@@ -0,0 +1,313 @@
+       Connection: keep-alive
+       Content-Length: 0
+       Host: localhost:5000
+       User-Agent: HTTPie/0.9.3
+
+    **Example response**
+
+    .. code-block:: http
+
+       HTTP/1.0 200 OK
+       Content-Length: 2
+       Content-Type: application/json
+       Date: Tue, 01 Mar 2016 17:21:28 GMT
+       Location: http://localhost:5000/builds/1
+       Server: Werkzeug/0.11.3 Python/3.5.0
+
+       {}
+
+    :reqheader Authorization: Include the token in a username field with a
+        blank password; ``<token>:``.
+    :param id: ID of the build.
+
+    :resheader Location: URL of the created build.
+
+    :statuscode 200: No error.
+    :statuscode 404: Build not found.
+    """
+    build = Build.query.get_or_404(id)
+    build.register_upload()
+    db.session.commit()
+    return jsonify({}), 200, {'Location': build.get_url()}
+
+
+@api.route('/builds/<int:id>', methods=['DELETE'])
+@token_auth.login_required
+def deprecate_build(id):
+    """Mark a build as deprecated (token required).
+
+    **Example request**
+
+    .. code-block:: http
+
+       DELETE /builds/1 HTTP/1.1
+       Accept: */*
+       Accept-Encoding: gzip, deflate
+       Authorization: Basic ZXlKcFlYUWlPakUwTlRZM056SXpORGdzSW1WNGNDSTZNVFEx...
+       Connection: keep-alive
+       Content-Length: 0
+       Host: localhost:5000
+       User-Agent: HTTPie/0.9.3
+
+    **Example response**
+
+    .. code-block:: http
+
+       HTTP/1.0 200 OK
+       Content-Length: 2
+       Content-Type: application/json
+       Date: Tue, 01 Mar 2016 17:21:29 GMT
+       Server: Werkzeug/0.11.3 Python/3.5.0
+
+       {}
+
+    :reqheader Authorization: Include the token in a username field with a
+        blank password; ``<token>:``.
+    :param id: ID of the build.
+
+    :statuscode 200: No error.
+    :statuscode 404: Build not found.
+    """
+    build = Build.query.get_or_404(id)
+    build.deprecate_build()
+    db.session.commit()
+    return jsonify({}), 200
+
+
+@api.route('/products/<slug>/builds/', methods=['GET'])
+def get_product_builds(slug):
+    """List all builds for a product.
+
+    **Example request**
+
+    .. code-block:: http
+
+       GET /products/lsst_apps/builds/ HTTP/1.1
+
+    **Example response**
+
+    .. code-block:: http
+
+       HTTP/1.0 200 OK
+       Content-Length: 96
+       Content-Type: application/json
+       Date: Tue, 01 Mar 2016 17:21:28 GMT
+       Server: Werkzeug/0.11.3 Python/3.5.0
+
+       {
+           "builds": [
+               "http://localhost:5000/builds/1",
+               "http://localhost:5000/builds/2"
+           ]
+       }
+
+    :param slug: Slug of the Product.
+
+    :>json array builds: List of URLs of Build entities for this product.
+
+    :statuscode 200: No error.
+    :statuscode 404: Product not found.
+    """
+    build_urls = [build.get_url() for build in
+                  Build.query.filter(Product.slug == slug).all()]
+    return jsonify({'builds': build_urls})
```

Comment:
Everything is technically correct with get_product_builds (/builds). But you are returning deprecated builds here too. I don't know if that'll be handled somewhere else? Or there could be more filtering here (possibly through parameters).